### DISCUSSION TITLE
[lodash] Support dot notation in lodash iteratee

### DISCUSSION THREAD

**Discussion by danvk** [DefinitelyTyped/DefinitelyTyped#66643] [Issues with a @types package]
Lodash supports dotted object notation for the "iteratee" argument to `_.map` and other functions:

```
> _.map([{name: {first: 'John', last: 'Doe'}, age: 42}], 'name.first')
['John']
```

The `@types/lodash` declarations don't seem to support this:

```ts
import _ from 'lodash';

declare const nested: {name: {first: string; last: string}; age: number; };
const firsts = _.map(nested, 'name.first');
//    ^? const firsts: any[]
```

Thanks to template literal types, this is [definitely possible](https://tsch.js.org/270/). It would be nice if the lodash typings had special support for this case.

[a comment by an automated account is not shown]

**Comment by aj-r** [comment]
We support this for `_.get`, so it should be feasible. We'd need to be careful not to break any existing use cases, though. I don't have time to work on this but would be happy to review a PR.

In the meantime, there are a few workarounds which I think should work. Now that TS supports the `?.` operator, I generally recommend that over lodash's string accessors, since the typings work more reliably, and should have slightly better performance.

```ts
array?.map(x => x?.name?.first); // (or remove the ? if you know it's always defined)
_.map(array, x => x?.name?.first);
_.map(array, ['name', 'first']);
```
